### DISCUSSION TITLE
Is it possible to define a reaction in ChemicalSystem after ChemicalState?

### DISCUSSION THREAD

**Discussion by hjunglab** [reaktoro/reaktoro#346] [Q&A]
### Description

I am going through the tutorial and wondering if it is possible to define a kinetic reaction after initiating a chemical state. In the tutorial, GeneralReaction is defined when initially defining a variable 'system'. What I am wondering is if it is possible to add a reaction after ChemicalSystem. 

### Example

~~~py
from reaktoro import *

system = ChemicalSystem(db,
    AqueousPhase("H2O H+ OH- Na+ Cl- NaOH").set(ActivityModelPhreeqc(db)),
    MineralPhase("Halite"),   
)

state = ChemicalState(system)
state.temperature(25.0, "C")
state.pressure(1.0, "bar")
state.set("H2O", 1.0, "kg")
state.scalePhaseVolume("Halite", 1.0, "cm3")  # start with 1 cm3 cube of halite crystal

system.GeneralReaction("Halite = Na+ + Cl-").setRateModel(ratefn)  # define the reaction and set its reaction rate model! # is there a way of doing something like this?
~~~

### What is your operating system?

macOS

**Comment by allanleal** [comment]
Hi @hjunglab, by design a `ChemicalSystem` object cannot be changed once it is constructed (it is immutable). Therefore, you need to specify the reactions controlled by kinetics when defining your chemical system.

It would be interesting to learn more about your use case that requires the reaction to be defined later rather than earlier. Maybe you are assuming some behavior that doesn't really exist. For example, kinetic reactions are not used when using `EquilibriumSolver` to perform chemical equilibrium calculations (say, in case you want to prepare an initial condition for a kinetics problem). For those reactions to be used, it would be necessary to perform chemical kinetics calculations using `KineticsSolver`.
